internal/session/command.go

```diff
@@ -37,43 +31,46 @@ func (s *Session) startCommandReader(ctx context.Context, del string) <-chan com
 			{0x16, 0x00, 0x00}, // 0.0
 		}
 
+		parser := command.NewParserWithLiteralContinuationCb(s.scanner, func() error { return response.Continuation().Send(s) })
+
 		for {
-			line, literals, err := s.liner.Read(func() error { return response.Continuation().Send(s) })
-			if err != nil {
-				return
-			}
+			s.inputCollector.Reset()
 
-			s.logIncoming(string(line), xslices.Map(maps.Keys(literals), func(k string) string {
-				return fmt.Sprintf("%v: '%s'", k, literals[k])
-			})...)
+			cmd, err := parser.Parse()
+			s.logIncoming(string(s.inputCollector.Bytes()))
+			if err != nil {
+				parserError, ok := err.(*rfcparser.Error) //nolint:errorlint
```

Comment:
I'd prefer you use errors.As here rather, any reason not to?

Reply:
Done